### ISSUE TITLE
Add an onError callback to Mutation component

### ISSUE THREAD

**Issue by kasonde** [zino-app/graphql-flutter#365]
**Describe the bug**
When using a mutation, I'm unable to tell whether there are mutation has errors or not in the onComplete callback of the mutation component. i.e

```
Mutation(
...,
onComplete: (result){
    // result is a json object. result.hasErrors doesn't work anymore
})
```

**Comment by mainawycliffe** [comment]
Yes, as currently  implemented, it returns the data only and not a `QueryResult` class which would expose the error. I am not sure why this was implemented this, so i need to do some more research.

**Comment by mainawycliffe** [comment]
After some initial research, the API works as expected returning data on complete, but it is missing `onError` which is a callback for catching when errors occur during mutation.

**Comment by kasonde** [comment]
@mainawycliffe  any idea of the progress of this? Do you have any workarounds?

**Comment by mainawycliffe** [comment]
@kasonde you can access errors on the `onUpdate` callback, which returns a `QueryResult` Object, check the example given [here](https://github.com/zino-app/graphql-flutter/blob/16fb1d507c44c24c0cf7da56dac304b956a95643/packages/graphql_flutter/example/lib/graphql_widget/main.dart#L211).

```dart
...
update: (Cache cache, QueryResult result) {
  if (result.hasErrors) {
    print(result.errors);
  } 
  ...
}
...
```

**Comment by mainawycliffe** [comment]
@micimize should we add an `OnError` method to be called in case of error, just like the `onCompleted` callback is called in case of data?

**Comment by micimize** [comment]
@mainawycliffe yes - appropriating this issue for that.

I think we should take the opportunity to implement an analogue to [`ApolloError`](https://github.com/apollographql/apollo-client/blob/7eaf4132cd2cd6244260777799406aaa03fcf377/packages/apollo-client/src/errors/ApolloError.ts#L33) as a step towards #319 (making it a breaking api change) and implement `onError(OurCustomErrorClass)`, bringing us closer to [apollo's `Mutation` props](https://www.apollographql.com/docs/react/essentials/mutations/#props)

Deciding on what to name the error is tough - maybe `GraphQLError` or `GraphQLErrorContainer`?

**Comment by mainawycliffe** [comment]
I will get on this over the weekend.

**Comment by mainawycliffe** [comment]
Closed by #435
